pgrx-tests/src/framework.rs

```diff
@@ -420,26 +466,42 @@ fn initdb(postgresql_conf: Vec<&'static str>) -> eyre::Result<()> {
 }
 
 fn modify_postgresql_conf(pgdata: PathBuf, postgresql_conf: Vec<&'static str>) -> eyre::Result<()> {
-    let mut postgresql_conf_file = std::fs::OpenOptions::new()
-        .write(true)
-        .truncate(true)
-        .open(format!("{}/postgresql.auto.conf", pgdata.display()))
-        .wrap_err("couldn't open postgresql.auto.conf")?;
-    postgresql_conf_file
-        .write_all("log_line_prefix='[%m] [%p] [%c]: '\n".as_bytes())
-        .wrap_err("couldn't append log_line_prefix")?;
+    let mut contents = String::new();
 
+    contents.push_str("log_line_prefix='[%m] [%p] [%c]: '\n");
+    contents
+        .push_str(&format!("unix_socket_directories = '{}'\n", pgdata.parent().unwrap().display()));
     for setting in postgresql_conf {
-        postgresql_conf_file
-            .write_all(format!("{setting}\n").as_bytes())
-            .wrap_err("couldn't append custom setting to postgresql.conf")?;
+        contents.push_str(&format!("{setting}\n"));
+    }
+
+    let postgresql_auto_conf = pgdata.join("postgresql.auto.conf");
+
+    if let Some(runas) = get_runas() {
+        let mut sudo_command = Command::new("sudo")
+            .arg("-u")
+            .arg(&runas)
+            .arg("tee")
+            .arg(postgresql_auto_conf)
```

Comment:
We use `sudo` at various points instead of setting the user more pervasively for our execution of these steps. Perhaps we should simply reexecute one of our commands but with `sudo` in order to accomplish this goal instead of fallibly setting it at every step? Would that work? It would be nice if this could be a design more resilient to how things change inside pgrx.

Reply:
It'd be nice if this would work, but it would require the user we're becoming to also have (at least) `cargo-pgrx` installed/available to them.  And if that were the case, then the caller could just run `sudo -u otheruser cargo pgrx test` directly.

We also have to su to `root` in a few places to get files copied.  So we become two different users, essentially.

Reply:
Ah, yeah, I hadn't parsed out the flow fully of having to jump between two users at that point, and still wasn't sure if my model was correct after I did... :sob: So much for that idea.